janitor/pkg/controller/terminatenode_controller.go

```diff
@@ -76,6 +167,46 @@ func (r *TerminateNodeReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	// Set the start time if it is not already set
 	terminateNode.SetStartTime()
 
+	// Check if max retries exceeded
+	if terminateNode.Status.RetryCount >= MaxTerminateRetries {
+		logger.Info("max retries exceeded, marking as failed",
+			"node", terminateNode.Spec.NodeName,
+			"retries", int(terminateNode.Status.RetryCount),
+			"maxRetries", MaxTerminateRetries)
+
+		terminateNode.SetCompletionTime()
+		terminateNode.SetCondition(metav1.Condition{
+			Type:   janitordgxcnvidiacomv1alpha1.TerminateNodeConditionNodeTerminated,
+			Status: metav1.ConditionFalse,
+			Reason: "MaxRetriesExceeded",
+			Message: fmt.Sprintf("Node failed to terminate after %d retries over %s",
+				MaxTerminateRetries, r.getTerminateTimeout()),
+			LastTransitionTime: metav1.Now(),
+		})
+
+		metrics.IncActionCount(metrics.ActionTypeTerminate, metrics.StatusFailed, terminateNode.Spec.NodeName)
+
+		result = ctrl.Result{} // Don't requeue
+
+		// Update status and return
+		if !reflect.DeepEqual(originalTerminateNode.Status, terminateNode.Status) {
+			var freshTerminateNode janitordgxcnvidiacomv1alpha1.TerminateNode
```

Comment:
You wouldn't need to get a fresh copy, no other entity should be updating the status, so we should be able to just apply the update and let the API server do it's thing